src/modules/csm/csm.py

```diff
@@ -110,18 +108,24 @@ def build_report(self, blockstamp: ReferenceBlockStamp) -> tuple:
         else:
             logger.info({"msg": "No previous CID available"})
 
-        tree = self.make_tree(shares)
-        tree_cid: CID | None = None
+        if distributed > 0:
+            curr_tree = self.make_tree(shares)
+            if not curr_tree:
+                raise InconsistentData("No tree to publish but shares are distributed")
```

Comment:
I'm not sure if it makes sense. `make_tree` doesn't depend on `distributed`. I think it's better to check on line 98 invariant `distributed and shares`.

Reply:
It's guaranteed by `calculate_distribution` by the way.

Reply:
changed